packages/pds/src/index.ts

```diff
@@ -91,9 +94,25 @@ export class PDS {
 
         return XRPCError.fromError(err)
       },
-      rateLimits: ctx.ratelimitCreator
+      rateLimits: rateLimits.enabled
         ? {
-            creator: ctx.ratelimitCreator,
+            creator:
+              rateLimits.mode === 'redis'
+                ? (opts) => RateLimiter.redis(ctx.redisScratch!, opts)
```

Comment:
What happens in case of a misconfiguration, where `rateLimits.mode === 'redis'` but `ctx.redisScratch` isn't configured?

Reply:
There is an assert in the config parsing but you're right